pkg/apis/operations/v1alpha1/imageprepull_types.go

```diff
@@ -0,0 +1,146 @@
+	// +optional
+	NodeNames []string `json:"nodeNames,omitempty"`
+	// LabelSelector is a filter to select member clusters by labels.
+	// It must match a node's labels for the NodeUpgradeJob to be operated on that node.
+	// Please note that sets of NodeNames and LabelSelector are ORed.
+	// Users must set one and can only set one.
+	// +optional
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+
+	// CheckItems specifies the items need to be checked before the task is executed.
+	// The default CheckItems value is disk.
+	// +optional
+	CheckItems []string `json:"checkItems,omitempty"`
+
+	// ImageSecret specifies the secret for image pull if private registry used.
+	// Use {namespace}/{secretName} in format.
+	// +optional
+	ImageSecret string `json:"imageSecrets,omitempty"`
+
+	// Concurrency specifies the max number of edge nodes that can be upgraded at the same time.
+	// The default Concurrency value is 1.
+	// +optional
+	Concurrency int32 `json:"concurrency,omitempty"`
+
+	// TimeoutSecondsOnEachNode limits the duration of the image prepull job on each edgenode.
+	// Default to 360.
+	// If set to 0, we'll use the default value 300.
+	// +optional
+	TimeoutSecondsOnEachNode *uint32 `json:"timeoutSecondsOnEachNode,omitempty"`
+
+	// RetryTimesOnEachNode specifies the retry times if image pull failed on each edgenode.
+	// Default to 0
+	// +optional
+	RetryTimesOnEachNode int32 `json:"retryTimesOnEachNode,omitempty"`
+}
+
+// PrepullState describe the PrepullState of image prepull operation on edge nodes.
+// +kubebuilder:validation:Enum=prepulling;successful;failed
+type PrepullState string
+
+// Valid values of PrepullState
+const (
+	PrepullInitialValue PrepullState = ""
+	Prepulling          PrepullState = "prepulling"
+	PrepullSuccessful   PrepullState = "successful"
+	PrepullFailed       PrepullState = "failed"
+)
+
+// ImagePrepullJobStatus stores the status of ImagePrepullJob.
+// contains images prepull status on multiple edge nodes.
+// +kubebuilder:validation:Type=object
+type ImagePrepullJobStatus struct {
+	// State represents for the state phase of the ImagePrepullJob.
+	// There are four possible state values: "", prechecking, prepulling, successful, failed.
+	State PrepullState `json:"state,omitempty"`
+
+	// Status contains image prepull status for each edge node.
+	Status []ImagePrepullStatus `json:"status,omitempty"`
+}
+
+// ImagePrepullStatus stores image prepull status for each edge node.
+// +kubebuilder:validation:Type=object
+type ImagePrepullStatus struct {
+	// NodeName is the name of edge node.
+	NodeName string `json:"nodeName,omitempty"`
+
+	// State represents for the state phase of the ImagePrepullJob on the edge node.
+	// There are five possible state values: "", prepulling, successful, failed.
+	State PrepullState `json:"state,omitempty"`
+
+	// Message represents for the status of each image pull on the edge node.
+	// It can be parsed into the format []ImageStatusEachNode.
```

Comment:
The `ImageStatusEachNode` should also be defined at here.

Reply:
> The `ImageStatusEachNode` should also be defined at here.

I have redefined the status and use `ImageStatus` struct to display the status of each image.